code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm

```diff
@@ -0,0 +1,460 @@
+	"<span class='highdanger'>The world around you reels, reality seeming to twist and tear until you find yourself trapped in a forsaken void beyond space and time.</span>")
+	playsound(banished_turf, 'sound/weapons/emitter2.ogg', 50, 1) //this isn't quiet
+
+	to_chat(TI,"<span class='xenodanger'>We have banished [banishment_target.name] to nullspace for [duration * 0.1] seconds.</span>")
+
+	addtimer(CALLBACK(src, .proc/banish_warning), duration * 0.7) //Warn when Banish is about to end
+	addtimer(CALLBACK(src, .proc/banish_deactivate), duration)
+
+	succeed_activate()
+	add_cooldown()
+
+	GLOB.round_statistics.wraith_banishes++
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "wraith_banishes") //Statistics
+
+/datum/action/xeno_action/activable/banish/proc/banish_warning()
+	var/mob/living/carbon/xenomorph/wraith/TI = owner
+
+	if(!TI) //Sanity
+		return
+
+	if(!banished_turf || !banishment_target)
+		return
+
+	to_chat(TI,"<span class='highdanger'>Our banishment target [banishment_target.name] is about to return!</span>")
+	owner.playsound_local(owner, 'sound/voice/hiss4.ogg', 50, 0, 1)
+
+
+/datum/action/xeno_action/activable/banish/proc/banish_deactivate()
+	SIGNAL_HANDLER
+	var/mob/living/carbon/xenomorph/wraith/TI = owner
+
+	if(!TI) //Sanity
+		return
+
+	if(!banished_turf || !banishment_target)
+		return
+
+	banishment_target.forceMove(banished_turf)
+	teleport_debuff_aoe(banishment_target) //Debuff/distortion when we reappear
+
+	banishment_target.visible_message("<span class='warning'>[banishment_target.name] abruptly reappears!</span>", \
+	"<span class='warning'>You suddenly reappear back in what you believe to be reality. It takes you a moment to regain your bearings.</span>")
+
+	if(isliving(banishment_target))
+		var/mob/living/L = banishment_target
+		L.Stun(1 SECONDS) //Short stun upon returning to reality
+
+	to_chat(TI, "<span class='xenodanger'>Our target [banishment_target.name] has returned to reality.</span>") //Always alert the Wraith
+
+	if(portal)
+		QDEL_NULL(portal) //Eliminate the Brazil portal if we need to
+
+	banishment_target = null //Clear vars
+	banished_turf = null
+	portal = null
+
+	return COMPONENT_BANISH_TARGETS_EXIST //For the recall sub-ability
+
+// ***************************************
+// *********** Recall
+// ***************************************
+/datum/action/xeno_action/recall
+	name = "Recall"
+	action_icon_state = "recall"
+	mechanics_text = "We recall a target we've banished back from the depths of nullspace."
+	use_state_flags = XACT_USE_NOTTURF //So we can recall ourselves from nether Brazil
+	cooldown_timer = 1 SECONDS //Token for anti-spam
+	keybind_signal = COMSIG_XENOABILITY_RECALL
+
+
+/datum/action/xeno_action/recall/action_activate()
+
+	if(!(SEND_SIGNAL(owner, COMSIG_XENOMORPH_WRAITH_RECALL) & COMPONENT_BANISH_TARGETS_EXIST))
+		to_chat(owner,"<span class='xenodanger'>We have no targets banished!</span>")
+		return fail_activate()
```

Comment:
What is this signal abuse and why does it exist

Reply:
So we don't have to reference the ability, and its vars directly via locate.

Reply:
Also I consulted with spooky, and he recommended this given what I was looking to achieve.

Reply:
ok